### PR TITLE
Add support for juju wallaby and python3

### PR DESCRIPTION
**Pull request by mdsufair**
Creating a new one after rebasing.

### CODE REVIEW

**Comment by mdsufair** [comment]
Created a new one after rebasing

File: shell_scripts/external_router_vms/tempest_conf.py [hunk 1]
```diff
@@ -106,13 +104,12 @@
 #admin_domain_scope = True
 uri = http://%(controller_ip)s:5000/v2.0/
 uri_v3 = http://%(controller_ip)s:5000/v3/
-#auth_version = v3
-auth_version = v2
+auth_version = v3
```
Comment (tbachman) [inline shell_scripts/external_router_vms/tempest_conf.py:107]: Currently, we need v2 still for queens. In June, we won't need that any more. The jenkins job for deploying the router is updating this template:
`````
# Tempest-specific fixes:
# 1) OSD Queens requires v3 keystone
YEAR=$(ssh noiro@${CTRL_SRVR} "ssh noiro@${RTR_IP} 'date'" | awk 'NF>1{print $NF}')
ssh noiro@${CTRL_SRVR} "echo ${YEAR} | sshpass -p noir0123 ssh -o StrictHostKeyChecking=no noiro@${RTR_IP} 'cat > ~/year.txt'"
if [ "${RELEASE}" = "queens" -o "${RELEASE}" = "train" ]; then
    CMD1="sed -i 's/auth_version = v2/auth_version = v3/g' ~/${YEAR}*/etc/tempest.conf >> /dev/null"
    CMD2="sed -i 's/api_v3 = false/api_v3 = true/g' ~/${YEAR}*/etc/tempest.conf >> /dev/null"
    echo '#!/bin/sh' | ssh noiro@${CTRL_SRVR} 'cat > /tmp/cmd.sh'
    echo ${CMD1} | ssh noiro@${CTRL_SRVR} 'cat >> /tmp/cmd.sh'
    echo ${CMD2} | ssh noiro@${CTRL_SRVR} 'cat >> /tmp/cmd.sh'
    ssh noiro@${CTRL_SRVR} "scp -o StrictHostKeyChecking=no /tmp/cmd.sh noiro@${RTR_IP}:~"
    ssh noiro@${CTRL_SRVR} "ssh -o StrictHostKeyChecking=no noiro@${RTR_IP} 'chmod +x ~/cmd.sh'"
    ssh noiro@${CTRL_SRVR} "ssh -o StrictHostKeyChecking=no noiro@${RTR_IP} '~/cmd.sh'"
    ssh noiro@${CTRL_SRVR} "rm -f /tmp/cmd.sh"
fi

I'd already made a change in run_tests.py to support changing the template for queens, so we probably should just set the keystone auth version as well, depending on the openstack version (v2 for queens, v3 after queens). In short, we can merge this, but we'll need to create a separate patch for run_tests.py to make that fix.

File: shell_scripts/external_router_vms/tempest_conf.py [hunk 2]
```diff
@@ -15,12 +15,10 @@
 
 AUTH_TEMPLATE="""
 [auth]
-# un-comment for juju queens
-#admin_domain_name = admin_domain
 tempest_roles = member
 admin_username = admin
 admin_project_name = admin
-admin_domain_name = Default
+admin_domain_name = %(admin_domain_name)s
```
Comment (tbachman) [inline shell_scripts/external_router_vms/tempest_conf.py:21]: Note: we'll have to update run_tests.py, as it's currently modifying this part of the template for queens (searches for the string 'admin_domain_name = Default' and deletes that line).

**Review by tbachman — COMMENTED** [review]
LGTM - just two things that we'll need to fix after merging.

**Pull request merged**